src/app/components/Notes/Modal/Modal.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react'
+import { ButtonsContainer, ColorContainer, ColorDateContainer, HeadingContainer, InputSectionContainer, MainContainer, NameContainer, DateContainer, Buttons, Label } from './ModalStyle'
+import { ModelOverlay } from '../NotesStyle';
+
+interface ModelProps{
+    closeModel:()=>void;
+    refresher:()=>void;
+    onDataUpdate:()=>void;
+    mode: string;
+    selectedNote: NoteItem | null;
+}
+
+interface NoteItem {
+    id: number;
+    title: string;
+    content: string;
+    background: string;
+    fontColor:string;
+    date: string;
+  }
+
+const Modal:React.FC<ModelProps>=({closeModel, refresher, onDataUpdate, mode, selectedNote })=> {
+    const [title, setTitle] = useState("");
+    const [content, setContent] = useState("");
+    const [background, setBackground] = useState("#ffffff");
+    const [fontColor, setfontColor] = useState("#ffffff");
+    const [date, setDate] = useState("");
+
+    useEffect(() => {
+      if (selectedNote) {
+        setTitle(selectedNote.title);
+        setContent(selectedNote.content);
+        setBackground(selectedNote.background);
+        setfontColor(selectedNote.fontColor);
+        setDate(selectedNote.date);
+      }
+    }, [selectedNote]);
+
+    const handleAdd = () => {
+      if (!title || !content || !date) {
+        return alert("Title, Content and Date are required");
+      } 
+      
+      const savedData: NoteItem[] = JSON.parse(localStorage.getItem("myNotes") || '[]');
+      if (mode === 'edit' && selectedNote) {
+        const updatedData = savedData.map((note) =>
+          note.id === selectedNote.id
+            ? {
+                ...note,
+                title,
+                content,
+                background,
+                fontColor,
+                date,
+              }
+            : note
+        );
+        localStorage.setItem("myNotes", JSON.stringify(updatedData));
+      } else {
+        const newData: NoteItem = {
+          id: Date.now(),
+          title,
+          content,
+          background,
+          fontColor,
+          date,
+        };
+        savedData.push(newData);
+        localStorage.setItem("myNotes", JSON.stringify(savedData));
+      }
+      setTitle("");
+      setContent("");
+      setDate("");
+      closeModel();
+      refresher();
+      onDataUpdate();
```

Comment:
why you need this? you are already calling refresher

Reply:
sorry it is a mistake i have removed the called function now.